### PR TITLE
Add restart to the launch config for nodejs projects

### PR DESCRIPTION
**Pull request by nturinski**
Fixes https://github.com/microsoft/vscode-azurefunctions/issues/781
Fixes https://github.com/microsoft/vscode-azurestaticwebapps/issues/573

Tracking for Python: https://github.com/microsoft/vscode-python/issues/15431

We discussed offline if we should retroactively ask users to alter their launch configurations for them, but we decided it was probably going to annoy most current users rather than help them.  It's pretty easy for a user to add the restart property to their own `launch.json`.

### CODE REVIEW

File: src/debug/NodeDebugProvider.ts [hunk 1]
```diff
@@ -15,7 +15,8 @@ export const nodeDebugConfig: DebugConfiguration = {
     type: 'node',
     request: 'attach',
     port: defaultNodeDebugPort,
-    preLaunchTask: hostStartTaskName
+    preLaunchTask: hostStartTaskName,
+    restart: true
```
Comment (ejizba) [inline src/debug/NodeDebugProvider.ts:19]: As discussed offline, "restart" doesn't seem to work on Mac. We'll need to figure out why that is before we can merge

**Comment by stefanushinardi** [comment]
@nturinski ping to check if there is any progress for this PR

**Comment by nturinski** [comment]
Using the latest func CLI tools (3.0.3331) and VS Code (1.58.0) I can confirm that this still doesn't work on MacOS.  I need to investigate why this isn't working

**Comment by alexweininger** [comment]
On Mac I'm only able to get "restart" to work when I run func start with the following arguments:

`func start --javascript --language-worker -- "--inspect=9229"`

**Comment by nturinski** [comment]
Confirmed that the workaround works on Windows as well though the command is `func start --javascript --language-worker="--inspect=9229"`

With this PR, we should change the `func host start` task to include those parameters so that "restart" will work on both Windows and Mac.

I'll have to do some more investigation, but we should probably omit the port number to the inspect flag.  The port number can be whatever the user defines it to be, and I believe that there is logic in the extension to retrieve the port number dynamically.  

I've confirmed that it works without it on Windows and this [workaround](https://github.com/Azure/azure-functions-core-tools/issues/1738#issuecomment-622553961) didn't include it.

**Comment by nturinski** [comment]
Closing due to lack of activity.

**Pull request closed without merging**